searx/plugins/engine_rate_limiter.py

```diff
@@ -0,0 +1,74 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+# pyright: basic
+"""Rate limit outgoing requests per engine
+
+Enable ``settings.yml``:
+
+- ``redis.url: ...`` check the value, see :ref:`settings redis`
+- ``rate_limit: ...`` max_requests and interval, as specified below
+- ``max_requests: ...`` max number of requests for that engine per interval
+- ``interval: ...`` number of seconds before rate limiting resets (optional, by default 1 second)
+"""
+
+from searx.engines import engines
+from searx.shared import redisdb
+
+name = "Engine rate limiter"
+description = "Limit the number of outgoing requests per engine"
+default_on = True
+preference_section = 'service'
+
+
+def check_rate_limiter(engine_name, limit, interval):
+    redis_client = redisdb.client()
+    lua_script = """
+    local engine = KEYS[1]
+    local limit = ARGV[1]
+    local interval = ARGV[2]
+
+    local count = redis.call('GET', engine)
+    if count and count > limit then
+        return count
+    else
+        local newCount = redis.call('INCR', engine)
+        if newCount == 1 then
+            redis.call('EXPIRE', engine, interval)
+        end
+        return newCount
+    end
+    """
+    script_sha = redis_client.script_load(lua_script)
```

Comment:
If `script_sha` is initialized in the `init` function, then this Python function doesn't have upload the script at each call.

Reply:
I am on the way to polish the implementation .. give me the time I need for.